crates/fuel-core/src/schema/tx/assemble_tx.rs

```diff
@@ -783,6 +803,9 @@ where
             *script.script_gas_limit_mut() = max_gas_limit;
 
             let (updated_tx, new_status) = self.arguments.dry_run(script).await?;
+
+            self.dry_run_count = self.dry_run_count.saturating_add(1);
```

Comment:
I think we should do a checked add here. I know we won't hit it in practice, but if we do it feels better to return an error than loop forever.

```suggestion
            self.dry_run_count = self.dry_run_count.checked_add(1).ok_or_else(|| anyhow::anyhow!("dry run count overflow"))?;
```